examples/sycl/pvc/pvc_gemm_with_epilogue_lincombdeeltact.cpp

```diff
@@ -0,0 +1,452 @@
+#include "cutlass/coord.h"
+
+#include "common.h"
+
+using namespace cute;
+
+///////////////////////////////////////////////////////////////////////////////////////////////////
+
+// Command line options parsing
+struct Options {
+
+  bool help;
+  bool error;
+
+  int m, n, k, l, iterations;
+  float alpha, beta;
+
+  Options():
+    help(false),
+    error(false),
+    m(5120), n(4096), k(4096), l(1), iterations(100),
+    alpha(1.f), beta(0.f)
+  { }
+
+  // Parses the command line
+  void parse(int argc, char const **args) {
+    cutlass::CommandLine cmd(argc, args);
+
+    if (cmd.check_cmd_line_flag("help")) {
+      help = true;
+      return;
+    }
+
+    cmd.get_cmd_line_argument("m", m, 5120);
+    cmd.get_cmd_line_argument("n", n, 4096);
+    cmd.get_cmd_line_argument("k", k, 4096);
+    cmd.get_cmd_line_argument("l", l, 1);
+    cmd.get_cmd_line_argument("alpha", alpha, 1.f);
+    cmd.get_cmd_line_argument("beta", beta, 0.f);
+    cmd.get_cmd_line_argument("iterations", iterations, 100);
+
+    // validate the arguments
+    bool m_valid = m > 0 and m % 8 == 0;
+    bool n_valid = n > 0 and n % 16 == 0;
+    bool k_valid = k > 0 and k % 16 == 0;
+    bool l_valid = l > 0;
+    if (!(m_valid and n_valid and k_valid and l_valid)) {
+      std::cout << "invalid arguments. Must be a multiple of (8, 16, 16)\n";
+      std::exit(1);
+    }
+  }
+
+  /// Prints the usage statement.
+  std::ostream & print_usage(std::ostream &out) const {
+
+    out << "PVC GEMM Example\n\n"
+      << "Options:\n\n"
+      << "  --help                      If specified, displays this usage statement\n\n"
+      << "  --m=<int>                   Sets the M extent of the GEMM\n"
+      << "  --n=<int>                   Sets the N extent of the GEMM\n"
+      << "  --k=<int>                   Sets the K extent of the GEMM\n"
+      << "  --l=<int>                   Sets the L extent (batch count) of the GEMM\n"
+      << "  --alpha=<s32>               Epilogue scalar alpha\n"
+      << "  --beta=<s32>                Epilogue scalar beta\n\n"
+      << "  --iterations=<int>          Iterations\n\n";
+
+    return out;
+  }
+};
+
+///////////////////////////////////////////////////////////////////////////////////////////////////
+
+template <typename T>
+struct sum_vals {
+  CUTLASS_HOST_DEVICE
+  T operator()(T a, T b) const {
+    return a + b;
+  }
+};
+
+template <typename T>
+struct diff_vals {
+  CUTLASS_HOST_DEVICE
+  T operator()(T a, T b) const {
+    return a - b;
+  }
+};
+
+template <typename Element>
+void print_device_tensor(DeviceAllocation<Element> device_alloc, std::size_t M, std::size_t N, std::size_t L, const char * message){
+    syclcompat::wait();
+    std::vector<Element> host_data(M*N*L);
+    device_alloc.copy_to_host(host_data.data());
+    auto tensor_aux_host = make_tensor(host_data.data(), cute::make_shape(M,N,L));
+    print(message);
+    print('\n');
+    print_tensor(tensor_aux_host);
+    print('\n');
+}
```

Comment:
this looks like debugging code. Do you need that in the final example?

Reply:
no, removed